concordium-smart-contract-testing/src/types.rs

```diff
@@ -600,3 +632,53 @@ pub struct ExchangeRateError;
 #[derive(Debug, Error)]
 #[error("Any signer must have at least one key.")]
 pub struct ZeroKeysError;
+
+/// Errors that occur while setting up the connection to an external node.
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum SetupExternalNodeError {
+    /// It was not possible to connect to a node on the provided endpoint.
+    #[error("Could not connect to the provided endpoint.")]
+    CannotConnect,
+    /// The endpoint provided was malformed.
+    #[error("The provided endpoint is malformed: '{endpoint}'.")]
+    MalformedEndpoint { endpoint: String },
+    /// Request timed out.
+    #[error("The request timed out.")]
+    Timeout,
+}
+
+/// Errors that occur while trying to communicate with an external node.
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum ExternalNodeError {
+    /// An external node has not been configured.
+    #[error("An external node has not been configured.")]
+    NotConfigured,
+    /// The query could not be performed.
+    #[error("Could not perform the query.")]
+    QueryError,
```

Comment:
Why does this not have any more details as to why it failed?

Reply:
Added now.